modules/oe_newsroom_newsletter/src/Api/NewsroomMessenger.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_newsroom_newsletter\Api;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\oe_newsroom\OeNewsroom;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * This connects to the Newsroom API and makes the requests to it using config.
+ *
+ * This class using the configuration page to configure itself automatically.
+ *
+ * @package Drupal\oe_newsroom_newsletter\Api
+ * @internal
+ */
+class NewsroomMessenger extends NewsroomMessengerBase {
+
+  /**
+   * Messenger constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Configuration factory to automatically load configurations.
+   * @param \Drupal\Core\Site\Settings $settings
+   *   Required for API private key.
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   Http client to send requests to the API.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, Settings $settings, ClientInterface $httpClient) {
+    $config = $configFactory->get(OeNewsroom::CONFIG_NAME);
+
+    $this->privateKey = $settings::get('newsroom_api_private_key');
+    $this->hashMethod = $config->get('hash_method');
+    $this->normalized = $config->get('normalized');
+    $this->universe = $config->get('universe');
+    $this->app = $config->get('app');
```

Comment:
This name is copied from API naming, it can be renamed to something different but for us is better to keep the same names. Do you want us to change it anyway?

Reply:
Yes, we wrap the API, we don't need to keep their names here.